downloads/swan-lake-release-notes/swan-lake-2201.3.0/RELEASE_NOTE.md

```diff
@@ -0,0 +1,460 @@
+- Improved data binding based on the MIME type match
+
+#### `graphql` package
+
+- Added service-level interceptor execution for records fields, maps, and tables
+- Added service-level interceptor execution for subscriptions
+- Changed to return all the errors related to a GraphQL document in a single response
+
+#### `gRPC` package
+
+- Updated Protocol Buffers version to 3.21.7
+
+### Bug fixes
+
+To view bug fixes, see the [GitHub milestone for 2201.3.0 (Swan Lake)](https://github.com/ballerina-platform/ballerina-standard-library/issues?q=is%3Aclosed+is%3Aissue+milestone%3A%222201.3.0%22+label%3AType%2FBug)
+
+## Developer tools updates
+
+### New features
+
+#### Language Server
+
+- Introduced a new `Extract to local variable` code action
+- Introduced a new `Extract to function` code action
+- Introduced the `loadProject()` API to the workspace manager to open up a project programmatically
+- Introduced a new `Generate module for client declaration` code action to generate a module for client declarations
+
+#### CLI
+
+##### Native-image build (Experimental)
+
+Introduced the `--native` flag, which generates a GraalVM native executable when building a Ballerina project.
+
+- The generated executable contains the modules in the current package, their dependencies, Ballerina runtime, and statically linked native code from the JDK
+
+- The new CLI command will be `bal build --native`
+
+>**Note:** There is a known [issue](https://github.com/ballerina-platform/ballerina-lang/issues/38665) in the native image build when using locks along with loops. 
+
+#### Debugger
+
+Introduced the `run-in-terminal` feature, which provides the capability to debug Ballerina programs (that take user inputs) in the launch mode. This can be achieved by setting the `terminal` attribute in the launch configurations to `integrated`. Once the configuration is set, launching the debugger will cause the Ballerina programs to be executed in a separate integrated VS Code terminal.
+
+### Improvements
+
+#### Language Server
+
+- Improved completion sorting within the expression of the `if` condition
+- Improved the `Change variable type` code action
+
+#### GraphQL Tool
+
+- Introduced single client generation for a single schema definition
+- Improved the GraphQL client config generation to support runtime configurability
+
+### Bug fixes
+
+To view bug fixes, see the GitHub milestone for 2201.3.0 (Swan Lake) of the repositories below.
+
+- [Language Server](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+milestone%3A2201.3.0+label%3AType%2FBug+label%3ATeam%2FLanguageServer)
+- [OpenAPI](https://github.com/ballerina-platform/openapi-tools/issues?q=is%3Aissue+milestone%3A%22Swan+Lake+2201.3.0%22+label%3AType%2FBug+is%3Aclosed)
+- [ProjectAPI](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+label%3ATeam%2FDevTools+milestone%3A2201.3.0+label%3AArea%2FProjectAPI)
```

Comment:
Shall we remove this line since there are no `project-api` bug fixes for update3

Reply:
Removed.